lib/datadog/core/diagnostics/environment_logger.rb

```diff
@@ -0,0 +1,282 @@
+# typed: true
+require 'date'
+require 'json'
+require 'rbconfig'
+
+module Datadog
+  module Core
+    module Diagnostics
+      # A holistic collection of the environment in which ddtrace is running.
+      # This logger should allow for easy reporting by users to Datadog support.
+      #
+      module EnvironmentLogger
+        class << self
+          # Outputs environment information to {Datadog.logger}.
+          # Executes only for the lifetime of the program.
```

Comment:
I guess this should be "executes only once..."?

Reply:
Yes